### ISSUE TITLE
clean old images out

### ISSUE THREAD

**Issue by paultag** [paultag/moxie#58]
purge out <none> images (new core?)

**Comment by tianon** [comment]
As an idea, maybe have the "pull" job go and inspect the image ID before
and after the pull, and if the new ID is different, go delete the old (so
that you're only cleaning up the specific images that moxie itself has left
dangling)?

**Comment by paultag** [comment]
@tianon yeah I thought of that too, only trouble is there might be other running jobs using the image, so we could still leak -- so I was thinking a new core to run; bad or good?

**Comment by tianon** [comment]
As long as the machine is just moxie, that doesn't seem crazy. :+1:

This is somewhere you have to be a little bit careful because the naive solution (`docker images -q --filter dangling=true | xargs docker rmi` or similar) can actually come back to bite you pretty hard because while a pull or build are in-progress, the layers are technically dangling, and Docker currently won't stop you from deleting them, which can cause fun errors, so you'd need to make sure that whatever does this blocks pulls while it's doing so (at least for now).

**Comment by paultag** [comment]
Ah, crap, good call. Hurm.

On Wed, Mar 25, 2015 at 5:54 PM, Tianon Gravi <notifications@github.com>
wrote:

> As long as the machine is just moxie, that doesn't seem crazy. [image:
> :+1:]
>
> This is somewhere you have to be a little bit careful because the naive
> solution (docker images -q --filter dangling=true | xargs docker rmi or
> similar) can actually come back to bite you pretty hard because while a
> pull or build are in-progress, the layers are technically dangling, and
> Docker currently won't stop you from deleting them, which can cause fun
> errors, so you'd need to make sure that whatever does this blocks pulls
> while it's doing so (at least for now).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paultag/moxie/issues/58#issuecomment-86230186>.
>



-- 
:wq

**Comment by paultag** [comment]
working around this with a cronjob
